### ISSUE TITLE
outgoing RPC bandwidth drops after a while, node becomes unconnectable via RPC

### ISSUE THREAD

**Issue by Gingeropolous** [monero-project/monero#7403]
I've noticed this with my high load public RPC servers. I monitor bandwidth with nload. Basically, when I start monerod, the outgoing bandwidth is high - for instance, right now I'm seeing ~100 Mbits / sec, and right now I have 53 RPC connections which I check with netstat -anp |grep 18089 |wc -l

 If it runs long enough, eventually nload will report that my outgoing bandwidth is maybe 200 kbits / second, and usually when this happens the RPC connection count is up to 100+. 

In these cases, when I restart monerod, the outgoing shoots right back on up to 100 Mbits / second.

I've also noticed that in these cases, if I try to connect to the node, my GUI can't make a connection. But once i restart the node, the GUI can connect. 

I've "solving" the problem currently with a cronjob, but... yeah.

**Comment by Gingeropolous** [comment]
was the same thing as #7482 

issue resolves by disabling rpc-ssl, so bug probably in that.

**Comment by hyc** [comment]
Agreed, a setting for max number of RPC clients has been needed anyway, just to control server load.

**Comment by Gingeropolous** [comment]
@vtnerd , 
> This seems like another issue entirely. SSL uses additional CPU load, so I wonder if this box is hitting that limit. 

The box is a 48 thread epyc with 64 GB ram, so I would hope thats not the case. Granted, its mining in the background with nice 19 , but i'm not sure if the other node operators that experienced this issue are *also* mining. And granted, I haven't tested if disabling mining makes the phenomenon go away. 

I'm pretty sure its the same issue as the SSL thing. When I posted this issue, I hadn't really started to dive into the problem. It's only when I started diving into the netstat output did I notice the CLOSE_WAIT state. 

but yeah, a max number of RPC clients would be good. Would be nice if the wallet / daemon had a handshake or whatever where if the daemon says it can't handle it, to get the wallet to re-request information from the DNS entry or whatever. I guess just initiate a new connection.
